concrete/elements/navigation/application_menu.php

```diff
@@ -3,45 +3,83 @@
+    <div class="dashboard-nav-rendered <?php if (isset($wrapperClass) && $wrapperClass) {
+        ?><?=$wrapperClass; ?><?php
+    } ?>">
         <?php if (isset($title) && $title) {
-            ?><h3><?=$title?></h3><?php
-        } ?>
+        ?><h3><?=$title; ?></h3><?php
+    } ?>
         <ul class="nav flex-column">
             <?php
-            $walk = function ($pages) use (&$walk, &$view) {
-                $n = count($pages);
-                if ($n > 0) {
-                    for ($i = 0; $i < $n; $i++) {
-                        $page = $pages[$i];
-                        $next = ($i + 1 < $n) ? $pages[$i + 1] : null;
-                        ?>
-                        <li class="<?=$view->controller->getMenuItemClass($page)?>">
-                            <a href="<?=$page->getCollectionLink()?>"><?=t($page->getCollectionName())?></a>
-                            <?php
-                            if ($view->controller->displayChildPages($page)) {
-                                $children = $view->controller->getChildPages($page);
-                                if (!empty($children)) {
-                                    ?>
-                                    <ul>
-                                    <?php $walk($children); ?>
-                                    </ul><?php
-                                }
-                            }
-                            ?>
+            $n = count($top);
+    if ($n > 0) {
+        for ($i = 0; $i < $n; ++$i) {
+            $page = $top[$i];
+            $next = ($i + 1 < $n) ? $top[$i + 1] : null;
+            $renderTopLevelMenu = $view->controller->displayTopLevelMenu($page);
+            $welcomeID = 185;
+            $c = Page::getCurrentPage(); ?>
+                <li class="<?=$view->controller->getMenuItemClass($page); ?>">
+                  <?php if ($renderTopLevelMenu == 1) {
+                ?>
+                    <a href="<?=$page->getCollectionLink(); ?>"><?=t($page->getCollectionName()); ?></a>
+                  <?php
+            } else {
+                ?>
+                    <a href="#" data-launch-sub-panel-url="<?=URL::to('/ccm/system/panels/dashboard/load_menu'); ?>"
+                    data-load-menu="<?=$page->getCollectionId(); ?>">
+                    <?=$page->getCollectionName(); ?>
+                    </a>
+                  <?php
+            } ?>
+                  <?php
+                    $myAccount = 192;
+            $editAccount = 193;
+            $waitingForMe = 186;
+            $waitingForMe = 186;
+            $messages = 195;
+            if (
+                      $c->getCollectionID() === $welcomeID ||
+                      $c->getCollectionID() === $myAccount ||
+                      $c->getCollectionID() === $waitingForMe ||
+                      $c->getCollectionID() === $editAccount ||
+                      $c->getCollectionID() === $messages
+                      ) {
+                if (
```

Comment:
what is the purpose of this double nested if statement?

Reply:
the first one checks against the current page, the second one checks against the menu item being looped through.

Reply:
Ive combined them into one if statement https://github.com/concrete5/concrete5/pull/8644/commits/d48a815e6be29f462737af505192aa2dbe46dcda